### ISSUE TITLE
Error starting Portable Version on Ubuntu 18.04.02 LTE

### ISSUE THREAD

**Issue by virivigio** [informatici/openhospital#18]
Hi
I was trying the PME release and just downloaded the zip file, extracted and run oh.sh
It reported an error with mysql

```
virgilio@virgilio-ubuntu:~$ cd Desktop/OH/
virgilio@virgilio-ubuntu:~/Desktop/OH$ ./oh.sh
Starting MySQL... 
Initializing database... on port 3307
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/home/virgilio/Desktop/OH/var/run/mysqld/mysql.sock' (2)
Error: Database not initialized!
virgilio@virgilio-ubuntu:~/Desktop/OH$ nohup: redirecting stderr to stdout
virgilio@virgilio-ubuntu:~/Desktop/OH$
```

anyway mysql, which is not installed in my system, remains running after the error

```
virgilio@virgilio-ubuntu:~/Desktop/OH$ ps -ef |grep mysql
virgilio  2658   906  0 17:51 ?        00:00:00 /bin/sh ./bin/mysqld_safe --defaults-file=/home/virgilio/Desktop/OH/etc/mysql/my.cnf
virgilio  2688  2658  0 17:51 ?        00:00:00 /home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686/bin/mysqld --defaults-file=/home/virgilio/Desktop/OH/etc/mysql/my.cnf --basedir=/home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686 --datadir=/home/virgilio/Desktop/OH/var/lib/mysql --pid-file=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.pid --skip-external-locking --port=3306 --socket=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.sock
virgilio  2689  2688  0 17:51 ?        00:00:00 /home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686/bin/mysqld --defaults-file=/home/virgilio/Desktop/OH/etc/mysql/my.cnf --basedir=/home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686 --datadir=/home/virgilio/Desktop/OH/var/lib/mysql --pid-file=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.pid --skip-external-locking --port=3306 --socket=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.sock
virgilio  2690  2689  0 17:51 ?        00:00:00 /home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686/bin/mysqld --defaults-file=/home/virgilio/Desktop/OH/etc/mysql/my.cnf --basedir=/home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686 --datadir=/home/virgilio/Desktop/OH/var/lib/mysql --pid-file=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.pid --skip-external-locking --port=3306 --socket=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.sock
virgilio  3324   906  0 18:07 pts/1    00:00:00 /bin/sh ./bin/mysqld_safe --defaults-file=/home/virgilio/Desktop/OH/etc/mysql/my.cnf
virgilio  3354  3324  0 18:07 pts/1    00:00:00 /home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686/bin/mysqld --defaults-file=/home/virgilio/Desktop/OH/etc/mysql/my.cnf --basedir=/home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686 --datadir=/home/virgilio/Desktop/OH/var/lib/mysql --pid-file=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.pid --skip-external-locking --port=3307 --socket=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.sock
virgilio  3355  3354  0 18:07 pts/1    00:00:00 /home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686/bin/mysqld --defaults-file=/home/virgilio/Desktop/OH/etc/mysql/my.cnf --basedir=/home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686 --datadir=/home/virgilio/Desktop/OH/var/lib/mysql --pid-file=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.pid --skip-external-locking --port=3307 --socket=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.sock
virgilio  3356  3355  0 18:07 pts/1    00:00:00 /home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686/bin/mysqld --defaults-file=/home/virgilio/Desktop/OH/etc/mysql/my.cnf --basedir=/home/virgilio/Desktop/OH/mysql-5.0.51a-linux-i686 --datadir=/home/virgilio/Desktop/OH/var/lib/mysql --pid-file=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.pid --skip-external-locking --port=3307 --socket=/home/virgilio/Desktop/OH/var/run/mysqld/mysql.sock
virgilio  3640  3277  0 18:18 pts/1    00:00:00 grep --color=auto mysql
virgilio@virgilio-ubuntu:~/Desktop/OH$ 

```

**Comment by virivigio** [comment]
Adding my.conf from mysql and database.properties from java application. but they look good

[my.cnf.txt](https://github.com/informatici/openhospital/files/3456513/my.cnf.txt)
[database.properties.txt](https://github.com/informatici/openhospital/files/3456514/database.properties.txt)

**Comment by virivigio** [comment]
windows version works

**Comment by pviotti** [comment]
Thanks for reporting this! Indeed the Linux version needs more testing. I'll have a look in the coming days.

**Comment by virivigio** [comment]
I cannot connect to db using mysqlworkbench: neither via tcp nor socket. root/empty password did not work, nor isf (and the password in the properties file)
but <oh folder>/var/log/mysql does not report any error opening the database or reading data files

**Comment by pviotti** [comment]
It looks like there are 2 issues: 
- it tries to open the MySQL socket file to initialize the database while the socket file does not exist yet. So a `while [ ! -f path/to/mysql.sock ]; do sleep 1; done` should fix it
- it uses java 6 32bit which does not work at all on my machine (and btw also MySQL looks like it's a 32bit version). I think we should stop supporting 32bit architectures since x64 is way more common nowadays.

I'll prepare these fixes in a couple of days and let you know when you can try it again. Thanks

**Comment by mwithi** [comment]
Mind that some features have only compliance with Java6 32bit and that the aim of the project is to apply also over obsolete machines.

Consider to prepare an additional 64bit package in place of update the current one.

**Comment by mwithi** [comment]
Mind that some features have only compliance with Java6 32bit and that the aim of the project is to apply also over obsolete machines.

Consider to prepare an additional 64bit package in place of update the current one.

**Comment by pviotti** [comment]
Right, let's keep the 32bit support (though we should be able to test it too..) and add the 64bit version.

Also, I wonder if by using a more recent version of the jre (jre 8, for instance) to execute the code compiled for java 6, the legacy native libraries would still work. Is there a way to test this?

**Comment by virivigio** [comment]
I took some more tests (virtualbox virtualization):
- as said ubuntu Desktop 64bit raises the error
- same error in ubuntu Desktop 32bit (see picture ubuntu32)
- windows 7 32 bit works fine
- windows 10 64 bit works fine
![ubuntu32](https://user-images.githubusercontent.com/1875906/62445375-5dcd7480-b760-11e9-8724-9965a23ab0ea.png)

quick note: on windows, clean machine, the first port used by db is 3307
![windowsports](https://user-images.githubusercontent.com/1875906/62445400-6f168100-b760-11e9-9226-dc2510efdabf.png)

if we need to test some specific os I'll be happy to help.

**Comment by pviotti** [comment]
The changes in PR #20 should fix the issues with Linux x32 and x64 (I've tested it with Arch x64 current and Ubuntu 16.04 x32).  
About the port used in Windows, it looks like it's by design (not sure why though) - see its [changelog](https://github.com/informatici/openhospital/blob/e1965d8afa4da57f3c95bf5d4ad165734cb469c5/poh-bundle-win/Changelog#L3).

**Comment by mwithi** [comment]
> About the port used in Windows, it looks like it's by design (not sure why though)
It's simply because to guess the port for the bundled MySQL is more complicated in windows batch files, though not impossible. If we want we can improve it.

**Comment by mwithi** [comment]
> Also, I wonder if by using a more recent version of the jre (jre 8, for instance) to execute the code compiled for java 6, the legacy native libraries would still work. Is there a way to test this?

It has been in the past a compatibility grid, never updated since then... Anyway, at that time, yet there were some compatibility issues for both 32bit and 64bit depending if on Unix or Windows OSs and java version, mostly concerning native libraries, so the work would be to redesign the involved features rather than update them; these are:
- webcam control
- dicom reader
- serial port communication
- txt/pos printer (not testest tested)
- xmpp protocol - internal chat (not tested yet)

**Comment by mwithi** [comment]
Closed with 9f46184f5c8822f117ab805dfc07dad74df0284d
